Tests/Serializer/BaseSerializerHandlerTest.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\CoreBundle\Tests\Serializer;
+
+use JMS\Serializer\GraphNavigator;
+use Sonata\CoreBundle\Tests\Fixtures\Bundle\Serializer\FooSerializer;
+
+/**
+ * @author Ahmet Akbana <ahmetakbana@gmail.com>
+ */
+final class BaseSerializerHandlerTest extends \PHPUnit_Framework_TestCase
+{
+    public function testGetSubscribingMethods()
+    {
+        $manager = $this->getMock('Sonata\CoreBundle\Model\ManagerInterface');
+
+        $serializer = new FooSerializer($manager);
+
+        $expectedMethods = array(
+            array(
+                'direction' => GraphNavigator::DIRECTION_SERIALIZATION,
+                'format' => 'json',
+                'type' => 'foo',
+                'method' => 'serializeObjectToId',
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_DESERIALIZATION,
+                'format' => 'json',
+                'type' => 'foo',
+                'method' => 'deserializeObjectFromId',
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_SERIALIZATION,
+                'format' => 'xml',
+                'type' => 'foo',
+                'method' => 'serializeObjectToId',
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_DESERIALIZATION,
+                'format' => 'xml',
+                'type' => 'foo',
+                'method' => 'deserializeObjectFromId',
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_SERIALIZATION,
+                'format' => 'yml',
+                'type' => 'foo',
+                'method' => 'serializeObjectToId',
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_DESERIALIZATION,
+                'format' => 'yml',
+                'type' => 'foo',
+                'method' => 'deserializeObjectFromId',
+            ),
+        );
+
+        $methods = $serializer->getSubscribingMethods();
+
+        $this->assertSame($methods, $expectedMethods);
```

Comment:
While writing the test for this class, I was aware that the formats were hardcoded. It is indeed weird. But since I was writing test, this was another concern. I don't want to change the structure of the class while writing test for it. This should be business of another PR IMO. After adding the property the tests also must be updated as the new structure requires.